Research/lva-node-app/public/js/main.js

```diff
@@ -0,0 +1,638 @@
+/**
+ * @fileoverview
+ * This file contains the majority of the functionality used across the various web pages, in the following order
+ * 1: universal functions
+ * 2: functions related to invoking methods
+ * 3: camera related functions
+ */
+
+
+//immutable global variables are written in all caps
+const PORT = 5000;
+
+/**
+ * @global variables. Mutable global variables in javascript are typically named in camelCase
+ */
+var graphTopologies={};
+var graphInstances={};
+var cameras={};
+
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// universal functions
+
+/**
+* get global variables from server, Returns a promise so that functions that depend on these variables can wait
+* @returns {Promise<any>}
+*/
+function getGlobals()
+{
+  return new Promise((resolve, reject) => 
+  {
+    var request = sendRequest("", `http://localhost:${PORT}/getglobals`, "GET");
+    request.onreadystatechange = function () 
```

Comment:
i think you have a race here... since sendRequest actually sends the request if it ever completed before you set this, it would never get called.

Reply:
same issue throughout this file

Reply:
see changes. Should be entirely resolved!